src/exotic-sizes.md

```diff
@@ -125,9 +153,30 @@ But this trick doesn't work yet.
 
 One final subtle detail about empty types is that raw pointers to them are
 actually valid to construct, but dereferencing them is Undefined Behavior
-because that doesn't actually make sense. That is, you could model C's `void *`
-type with `*const Void`, but this doesn't necessarily gain anything over using
-e.g. `*const ()`, which *is* safe to randomly dereference.
+because that wouldn't make sense.
+
+We recommend against modelling C's `void*` type with `*const Void`.
+A lot of people started doing that but quickly ran into trouble because
+people want to convert raw pointers to references when doing FFI, and making an
+`&Void` is Undefined Behaviour. `*const ()` (or equivalent) works just as well,
```

Comment:
Well, is it? It might be, it might not be. It is certainly a footgun though.

Reply:
The rust compiler literally emits a warning that your code is unreachable, and then your program SIGILL's.

Reply:
I feel like this is a bug in rustc...

Reply:
How are you making the `&Void`?

Reply:
https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=23236a60b8801319313ff8d475fdbb0e

MIRI also aborts for hitting unreachable code

Reply:
I think it's safe to conservatively call this UB in the nomicon and then soften the language if y'all ever settle on something more precise/subtle

Reply:
Yeah well you are constructing a `!` here and taking its address. Of course that does not work, and never will.

I thought of something more like https://play.rust-lang.org/?version=nightly&mode=release&edition=2018&gist=d4002e1c51c73aa38eba6f46afddab3a